### PR TITLE
Update test connection

### PR DESCRIPTION
**Pull request by MilanBariya**
### Describe your changes :

I worked on the updating test connection function for Bigquery.
Fix: #10144

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Review by NiharDoshi99 — APPROVED** [review]

File: ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py [hunk 1]
```diff
@@ -338,3 +338,11 @@ def close(self):
...
             os.unlink(self.temp_credentials)
         os.environ.pop("GOOGLE_CLOUD_PROJECT", "")
+
+    def test_connection(self) -> str:
+        """
+        Used a timed-bound function to test that the engine
+        can properly reach the source
+        """
+        test_connection_fn = get_test_connection_fn(self.service_connection)
+        test_connection_fn(self.engine, self.service_connection)
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py:348]: should this return something?

File: ingestion/src/metadata/ingestion/source/database/bigquery/connection.py [hunk 2]
```diff
@@ -74,8 +81,60 @@ def get_connection(connection: BigQueryConnection) -> Engine:
     )
 
 
-def test_connection(engine: Engine) -> None:
+def test_connection(engine: Engine, service_connection) -> str:
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/database/bigquery/connection.py:84]: is this the correct type? Or are we returning a `TestStepResult`?

File: ingestion/src/metadata/ingestion/connections/test_connections.py [hunk 3]
```diff
@@ -72,7 +76,9 @@ def test_connection_steps(steps: List[TestConnectionStep]) -> str:
             step.function()
             test_connection_result.success.append(f"'{step.name}': Pass")
 
-        except Exception:
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.debug(f"{step.name}-{exc}")
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/connections/test_connections.py:81]: should we have this as warning / error ?

Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/connections/test_connections.py:81]: L81

File: ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py (hunk 1 above)
Comment (MilanBariya) [reply ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py:348]: It should return None.
I will change this.

File: ingestion/src/metadata/ingestion/source/database/bigquery/connection.py (hunk 2 above)
Comment (MilanBariya) [reply ingestion/src/metadata/ingestion/source/database/bigquery/connection.py:84]: It should not be str, i will change this too.

File: ingestion/src/metadata/ingestion/connections/test_connections.py (hunk 3 above)
Comment (MilanBariya) [reply ingestion/src/metadata/ingestion/connections/test_connections.py:81]: Yes, warning makes sense here.

**Review by pmbrull — COMMENTED** [review]
thanks for the changes

**Review by pmbrull — APPROVED** [review]

**Pull request merged**